table/tables/partition.go

```diff
@@ -0,0 +1,169 @@
+	pi := tblInfo.GetPartitionInfo()
+	ctx := mock.NewContext()
+	partitionPruneExprs := make([]expression.Expression, 0, len(pi.Definitions))
+	locateExprs := make([]expression.Expression, 0, len(pi.Definitions))
+	var buf bytes.Buffer
+	for i := 0; i < len(pi.Definitions); i++ {
+		if strings.EqualFold(pi.Definitions[i].LessThan[0], "MAXVALUE") {
+			// Expr less than maxvalue is always true.
+			fmt.Fprintf(&buf, "true")
+		} else {
+			fmt.Fprintf(&buf, "((%s) < (%s))", pi.Expr, pi.Definitions[i].LessThan[0])
+		}
+		expr, err := expression.ParseSimpleExpr(ctx, buf.String(), tblInfo)
+		if err != nil {
+			// If it got an error here, ddl may hang forever, so this error log is important.
+			log.Error("wrong table partition expression:", errors.ErrorStack(err), buf.String())
+			return nil, errors.Trace(err)
+		}
+		locateExprs = append(locateExprs, expr)
+
+		if i > 0 {
+			fmt.Fprintf(&buf, " and ((%s) >= (%s))", pi.Expr, pi.Definitions[i-1].LessThan[0])
+		}
+
+		expr, err = expression.ParseSimpleExpr(ctx, buf.String(), tblInfo)
+		if err != nil {
+			// If it got an error here, ddl may hang forever, so this error log is important.
+			log.Error("wrong table partition expression:", errors.ErrorStack(err), buf.String())
+			return nil, errors.Trace(err)
+		}
+		partitionPruneExprs = append(partitionPruneExprs, expr)
+		buf.Reset()
+	}
+	return &PartitionExpr{
+		PartitionPrune: partitionPruneExprs,
+		Locate:         locateExprs,
+	}, nil
+}
+
+// PartitionExpr returns the partition expression.
+func (t *PartitionTable) PartitionExpr() *PartitionExpr {
+	return t.partitionExpr
+}
+
+func partitionRecordKey(pid int64, handle int64) kv.Key {
+	recordPrefix := tablecodec.GenTableRecordPrefix(pid)
+	return tablecodec.EncodeRecordKey(recordPrefix, handle)
+}
+
+// locatePartition returns the partition ID of the input record.
+func (t *PartitionTable) locatePartition(ctx sessionctx.Context, pi *model.PartitionInfo, r []types.Datum) (int64, error) {
+	var err error
+	partitionExprs := t.partitionExpr.Locate
+	idx := sort.Search(len(partitionExprs), func(i int) bool {
+		var ret int64
+		ret, _, err = partitionExprs[i].EvalInt(ctx, types.DatumRow(r))
+		if err != nil {
+			return true // Break the search.
+		}
+		return ret > 0
+	})
+	if err != nil {
+		return 0, errors.Trace(err)
+	}
+	if idx < 0 || idx >= len(partitionExprs) {
+		// The data does not belong to any of the partition?
+		return 0, errors.Trace(table.ErrTrgInvalidCreationCtx)
+	}
+	return pi.Definitions[idx].ID, nil
+}
+
+// GetPartition returns a Table, which is actually a Partition.
+func (t *PartitionTable) GetPartition(pid int64) table.Table {
+	var ret Partition
+	// Make a shallow copy, change ID to partition ID.
+	ret.tableCommon = t.tableCommon
+	ret.partitionID = pid
+	ret.recordPrefix = tablecodec.GenTableRecordPrefix(pid)
+	ret.indexPrefix = tablecodec.GenTableIndexPrefix(pid)
+	return &ret
+}
+
+// AddRecord overrides the AddRecord method for the table.Table interface.
```

Comment:
You can not override the method of an interface. Because it is not implemented in the interface. I guess you mean the method in tableCommon.

Reply:
So I'll change it to `implements`, Go has no "override" or other OO concept